scripts/globals/items/haste_belt.lua

```diff
@@ -3,25 +3,29 @@
 -- Item: Haste Belt
 -- Item Effect: 10% haste
 -----------------------------------
+
 local itemObject = {}
 
 itemObject.onItemCheck = function(target)
-    local effect = target:getStatusEffect(xi.effect.HASTE)
-    if effect ~= nil and effect:getItemSourceID() == xi.items.HASTE_BELT then
-        target:delStatusEffect(xi.effect.HASTE)
+    if target:getStatusEffect(xi.effect.ENCHANTMENT, nil, xi.items.HASTE_BELT) ~= nil then
+        target:delStatusEffect(xi.effect.ENCHANTMENT, nil, xi.items.HASTE_BELT)
     end
 
     return 0
 end
 
 itemObject.onItemUse = function(target)
     if target:hasEquipped(xi.items.HASTE_BELT) then
-        if not target:hasStatusEffect(xi.effect.HASTE) then
-            target:addStatusEffect(xi.effect.HASTE, 1000, 0, 180, 0, 0, 0, xi.items.HASTE_BELT)
-        else
-            target:messageBasic(xi.msg.basic.NO_EFFECT)
-        end
+        target:addStatusEffect(xi.effect.ENCHANTMENT, 0, 0, 180, 0, 0, 0, xi.items.HASTE_BELT)
     end
 end
 
+itemObject.onEffectGain = function(target, effect)
+    target:addMod(xi.mod.HASTE_MAGIC, 1000)
+end
+
+itemObject.onEffectLose = function(target, effect)
+    target:delMod(xi.mod.HASTE_MAGIC, 1000)
+end
+
```

Comment:
I don't believe you need this section.

Reply:
I'm not sure how it works. If I remove those lines the enchantment doesn't give any haste that I can see.

Reply:
Interesting! 
I'm going off of other similar usable items, like memento_muffler.lua - where the enchantment is applied in onItemUse and doesn't make use of onEffectGain / onEffectLose

Reply:
I think the problem was it was overwriting the spell Haste. For the most part memento_muffler.lua is written how haste_belt.lua was originally. I believe this line

// remove effects with same type
DelStatusEffectsByType(PStatusEffect->GetType());

in status_effect_container.cpp removes duplicate haste effects.

Reply:
Which is interesting, as when you give an enchantment with a specific ID using the haste belt, it shouldn't be overwritten.. hmm

Reply:
You know what? I'm totally wrong! You're doing this perfectly.!